src/udiskslinuxdrive.c

```diff
@@ -50,6 +51,71 @@
 #include "udisksdaemonutil.h"
 #include "udiskslinuxdevice.h"
 
+static const gchar *
+_get_blk_path (UDisksDrive *drv, GDBusMethodInvocation *invocation);
+
+static gboolean
+handle_turn_ident_ledon (UDisksDrive *_drive,
+                         GDBusMethodInvocation *invocation);
+
+static gboolean
+handle_turn_ident_ledoff (UDisksDrive *_drive,
+                          GDBusMethodInvocation *invocation);
+
+static gboolean
+handle_turn_fault_ledon (UDisksDrive *_drive,
+                         GDBusMethodInvocation *invocation);
+
+static gboolean
+handle_turn_fault_ledoff (UDisksDrive *_drive,
+                          GDBusMethodInvocation *invocation);
+
+#define _LED_FUN_GEN(func_name, lsm_func) \
```

Comment:
I'm sorry, but this is really ugly. If nothing else, it would cause ``gcc`` to generate weird errors 100 lines off etc. Why not to define a single function taking another function as one of its arguments and use that to implement the four ``handle_`` functions? The *btrfs* module already uses that trick, take a look at ``btrfs_subvolume_perform_action`` in *modules/btrfs/udiskslinuxfilesystembtrfs.c*.